File: src/Test/LazyEvaluationFFI.php
<?php
$exports['runLazyEvaluationFFI'] = function($count) {
    // Tail-recursive construction/repetition become loops; each force still
    // evaluates the full chain of 1000 closures, as in Test.LazyEvaluation.
    $result = 0;
    for ($repetition = 0; $repetition < (int)$count; $repetition++) {
        $thunk = function() { return 0; };
        for ($depth = 0; $depth < 1000; $depth++) {
            $previous = $thunk;
            $thunk = function() use ($previous) { return $previous() + 1; };
        }
        $result += $thunk();
    }
    return $result;
};
return $exports;

File: scratch/gopurs-adt-reuse-validation-20260916/after-fficc/src/Test/LazyEvaluationFFI.fs
module Test.LazyEvaluationFFI

type Thunk = unit -> int
let buildThunks depth (initial: Thunk) =
    let mutable result = initial
    for _ in 1 .. depth do
        let previous = result
        result <- fun () -> previous () + 1
    result
let runLazyEvaluationFFI (input: obj) =
    let mutable result = 0
    for _ in 1 .. unbox<int> input do
        result <- result + (buildThunks 1000 (fun () -> 0)) ()
    result :> obj
